packages/carousel/src/Carousel/Carousel.tsx

```diff
@@ -10,30 +10,34 @@ import {
   SlideProps,
 } from './types';
 
-export const Carousel: CarouselProps = ({ children, showDots = true, showArrows = false, ...props }) => {
+export const Carousel: CarouselProps = ({ children, showDots = true, showArrows = false, visibleSlides, ...props }) => {
   const [totalSlides] = Children.map(children, (child) => {
     if (isValidElement(child) && child.type === Carousel.Slider) {
       return Children.count(child.props.children) as number;
     }
   });
   if (!totalSlides) return;
 
-  const renderDots = (totalSlides: number) => {
+  const renderDots = (totalSlides: number, visibleSlides: number) => {
     const carouselDots: ReactNode[] = [];
-    for (let i = 0; i < totalSlides; i++) {
-      carouselDots.push(<Carousel.Dot key={i} slide={i} />);
+    const totalSlideGroups = Math.ceil(totalSlides / visibleSlides);
+
+    for (let i = 0; i < totalSlideGroups; i++) {
+      const slide = i * visibleSlides;
```

Comment:
What does it do?

Reply:
It sets the slide index to which the carousel should move to.

For example. On a list of 13 slides and 4 visible slides at a time, 4 dots will be rendered and their slide prop will be set to indexes: 0, 4, 8, 12.

In other words, the dot is set to the first slide in each group. That is what that variable is for.

You can test and preview that on this link
https://5eed577398d08d0022b52d76-oszkynchge.chromatic.com/?path=/story/carousel-carousel--playground&args=visibleSlides:4;totalSlides:13